### DISCUSSION TITLE
Render a bag of content items

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#8941] [Q&A] [answered]
How to render a bag of content items?

Imagine that we have
* Content-CarInfo.cshtm
* Section-TextBlock1.cshtm
* Section-ImagesBlock1.cshtm
* BlockItem-Image.cshtm

Car info contains a bag of sections.
Images block section contains a bag of image blok items

How to name and render that correctly?

**Comment by dafergu2** [comment]
```liquid  
{%- for item in Model.ContentItem.Content.BagPart.ContentItems -%}
    {{ item | shape_build_display: "Detail" | shape_render }}
{%- endfor -%}
```
I use the above code to render the bag of content items in my parent content type template.  This allows you to use the shape defined for the content item.  If the rendering of the bag items is specific to the parent container, then you can simply just use the content data from the item var instead of rendering it via `shape_render`.

If you're using Razor, you'll need to do it slightly different.  Copied from the url below and modified slightly...

```Razor
@*
Original: @using OrchardCore.Flows.ViewModels
Original: @model BagPartViewModel
*@

@inject OrchardCore.ContentManagement.Display.IContentItemDisplayManager ContentItemDisplayManager

@* Added *@
@inject OrchardCore.DisplayManagement.ModelBinding.IUpdateModelAccessor UpdateModelAccessor

<section class="flow">
    @foreach (var item in Model.BagPart.ContentItems)
    {
        // Original var itemContent = await ContentItemDisplayManager.BuildDisplayAsync(item,
        //      Model.BuildPartDisplayContext.Updater, Model.Settings.DisplayType ?? "Detail", Model.BuildPartDisplayContext.GroupId);
        var itemContent = await ContentItemDisplayManager.BuildDisplayAsync((ContentItem) item,
                UpdateModelAccessor.ModelUpdater, "Detail");
        @await DisplayAsync(itemContent)
    }
</section>
```

Please note that in the above code, I made some modifications that make sense to me, but you may not need them.  Full disclosure, I have not tested the code and if you have a Model you may want to use that directly.  See the original code on the page below:

https://docs.orchardcore.net/en/dev/docs/reference/modules/Flow/BagPart/

**Comment by kdubious** [comment, marked as answer]
Content-CarInfo.cshtml
`@await DisplayAsync(Model.Content.Sections)`
^ assuming the bag is named "Sections", and you just want to place the content of the bag somewhere

Content-CarInfo.cshtml
`@await DisplayAsync(Model.Content)`
^ assuming you want to render "all" of the CarInfo


Content-Section.Summary.cshtml
`<span>@Model.ContentItem.DisplayText</span>`
^ to show the title of the Section in a span

**Reply by xperiandri** [reply]
Got it, so the trick is `.Summary` suffix

**Reply by kdubious** [reply]
I think the "list" is by default "Summary," and if you render an individual item, it's "Detail."
